src/components/LmsConfigurations/CornerstoneIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,256 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+import { handleErrors, validateLmsConfigForm } from './common';
+
+export const REQUIRED_CORNERSTONE_CONFIG_FIELDS = [
+  'cornerstoneBaseUrl',
+];
+
+function configFormReducer(state, action) {
+  switch (action.type) {
+    case 'PENDING': {
+      return { ...state, error: null, submitState: SUBMIT_STATES.PENDING };
+    }
+    case 'ERROR': {
+      return { ...state, error: action.value, submitState: SUBMIT_STATES.ERROR };
+    }
+    case 'INVALID': {
+      return {
+        ...state,
+        submitState: SUBMIT_STATES.ERROR,
+        invalidFields: action.value,
+        error: 'Form could not be submitted as there are fields with invalid values. Please correct them below.',
+      };
+    }
+    case 'COMPLETE': {
+      return {
+        ...state,
+        invalidFields: {},
+        error: null,
+        submitState: SUBMIT_STATES.COMPLETE,
+      };
+    }
+    case 'SET_CONFIG': {
+      return {
+        ...state,
+        config: action.value,
```

Comment:
should setting the config make error states go away?

Reply:
It could, but the flow should be as follows:
Click submit > trigger "PENDING" state (which sets errors to null) > hit validation failure > error state is set > user fixes and hits submit again > trigger PENDING state again (thus clearing errors), etc.

So every time they hit submit we're clearing the errors so clearing it here would be a bit redundant.